Introduce `NetworkConfig` API (#449)
bitmovin/bitmovin-player-react-native: CHANGELOG.md

# Changelog

## [Unreleased]

### Added

- `NetworkConfig` to configure the network request manipulation functionality

>   - `preprocessHttpRequest(type, request) => Promise<HttpRequest>` to pre-process an HTTP request before the request is made

Reviewer: ```suggestion
  - `preprocessHttpRequest(type, request) => Promise<HttpRequest>` to pre-process an HTTP request before the request is executed by the player
```

>   - `preprocessHttpResponse(type, response) => Promise<HttpResponse>` to pre-process an HTTP response before it gets into the player

Reviewer: ```suggestion
  - `preprocessHttpResponse(type, response) => Promise<HttpResponse>` to pre-process an HTTP response before gets passed to the player
```
